src/Mutator/NodeMutationGenerator.php

```diff
@@ -86,27 +85,21 @@ public function __construct(
     /**
      * @return Mutation[]
      */
-    public function generate(Node $node): array
+    public function generate(Node $node): iterable
     {
-        return array_reduce(
-            $this->mutators,
-            function (array $mutations, Mutator $mutator) use ($node): array {
-                return $this->generateForMutator($node, $mutator, $mutations);
-            },
-            []
-        );
+        yield from take($this->mutators)->map(function (Mutator $mutator) use ($node) {
```

Comment:
I feel like in this case `nikic/iter` would be a better fit here:

```php
yield from map(
    map(function () { ... },
    $this->mutators
);
```

I'm checking the nikic alternative elsewhere: it doesn't make sense to use pipe _and_ iter. But I admit I have a liking for iter whenever dealing with iterables

Reply:
`nikic/iter` isn't solving the problem of `array_map` hell. E.g it doesn't have a fluent interface. It's a deal breaker for me.